plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -471,6 +483,48 @@ public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoCol
         return iterable.iterator();
     }
 
+    private static Set<String> findDbRefColumns(MongoCollection<Document> collection)
+    {
+        MongoCursor<Document> mongoCursor = collection.find().collation(SIMPLE_COLLATION).iterator();
+        if (mongoCursor.hasNext()) {
+            return mongoCursor.next().entrySet().stream()
+                    .filter(entry -> entry.getValue() instanceof DBRef)
+                    .map(entry -> entry.getKey())
+                    .collect(toImmutableSet());
+        }
+        return ImmutableSet.of();
+    }
+
+    /**
+     * Creates a set of sufficient columns for the input projected columns. For example,
+     * if input {@param columns} include columns "a.b" and "a.b.c", then they will be projected from a single column "a.b".
+     */
+    private static List<String> getSufficientColumns(List<MongoColumnHandle> columnHandles)
```

Comment:
Instead of having it as `String` can we have them as parts so that we could idenitfy the parent based on the index instead of parsing them based `.`